src/test/java/jewellery/inventory/integration/ResourceInUserCrudIntegrationTest.java

```diff
@@ -0,0 +1,386 @@
+
+import static jewellery.inventory.helper.ResourceTestHelper.getGemstoneRequestDto;
+import static jewellery.inventory.helper.UserTestHelper.createResourceInUserRequestDto;
+import static jewellery.inventory.helper.UserTestHelper.createTestUserRequest;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+import java.util.List;
+import java.util.UUID;
+import jewellery.inventory.dto.request.UserRequestDto;
+import jewellery.inventory.dto.request.resource.ResourceInUserRequestDto;
+import jewellery.inventory.dto.request.resource.ResourceRequestDto;
+import jewellery.inventory.dto.response.UserResponseDto;
+import jewellery.inventory.dto.response.resource.GemstoneResponseDto;
+import jewellery.inventory.dto.response.resource.ResourceInUserResponseDto;
+import jewellery.inventory.dto.response.resource.ResourceResponseDto;
+import jewellery.inventory.repository.ResourceInUserRepository;
+import jewellery.inventory.repository.ResourceRepository;
+import jewellery.inventory.repository.UserRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+public class ResourceInUserCrudIntegrationTest {
+  @Autowired TestRestTemplate testRestTemplate;
+  @Autowired UserRepository userRepository;
+  @Autowired ResourceRepository resourceRepository;
+  @Autowired ResourceInUserRepository resourceInUserRepository;
+
+  @Value(value = "${local.server.port}")
+  private int port;
+
+  private String getBaseResourceAvailabilityUrl() {
+    return "http://localhost:" + port + "/resources/availability";
+  }
+
+  private String getResourceAvailabilityUrl(UUID userId, UUID resourceId) {
+    return "http://localhost:" + port + "/resources/availability/" + userId + "/" + resourceId;
+  }
+
+  private String getBaseUserUrl() {
+    return "http://localhost:" + port + "/users";
+  }
+
+  private String getBaseResourceUrl() {
+    return "http://localhost:" + port + "/resources";
+  }
```

Comment:
Lots of hardcoded strings repetitions. Could we extract at least getBaseUrl() to return "http://localhost:" + port

Reply:
Fixed